nomenclature/processor/region.py

```diff
@@ -394,6 +407,20 @@ def apply(self, df: IamDataFrame, dsd: DataStructureDefinition) -> IamDataFrame:
                             if var not in vars_default_args
                         }
                         for cr in self.mappings[model].common_regions:
+                            # If the common region is only comprised of a single model
+                            # native region, just rename
+                            if (
```

Comment:
Thinking about this again, I think it would be preferable to only check for `is_single_constituent_region` because
1. first filtering to check if the region exists, then filtering again as part of the renaming is less performant than doing the renaming directly
2. the could would be more readable

If this creates verbose log output, you could use the `adjust_log_level` context
https://github.com/IAMconsortium/pyam/blob/de6b47bb1ea7e937ad32dc8a6c284beddca98685/pyam/logging.py#L9

Reply:
The reason I did it is directly related to iamconsortium/pyam#631, if the data frame that we would rename is empty, we get an error.
I totally see your point that this way is quite inefficient so what I could do alternatively is tackle the pyam issue first an then come back to this one.

Reply:
Thanks for the quick PR, removed the second condition after fixing the version requirement in setup.cfg